TakingITGlobal/commit2act
# [TIG-735] Mocked form fields and data interaction for InfoSecondary page

## Ticket(s)

* [TIG-735](https://app.clickup.com/t/9009201449/TIG-735) - Primary sub-task
* [TIG-732](https://app.clickup.com/t/9009201449/TIG-732) - Parent task

## Description

This PR mocks-in form fields with state controls, loading toggle, and a mocked API request method via hook for the `InfoSecondary` page. Fields and layout are not styled here, as that is intended to come with later styling/theming pass(es). Additionally, this PR sets up a pattern for rendering a submit button on pages with forms as well as redirecting to the next page after the API request returns successfully.

## Testing

**Note:** Please follow the order described, as tests do not currently pass when the `.env` var is changed

1. Pull down and install the branch (`npm install`)
2. Run all tests (`npm test`)
3. Update the `REACT_APP_SIGNUP_FLOW` value to `onboard` in the root `.env` file
4. Run the app locally (`npm start`)
5. Login with a test user
6. Navigate to `/onboard/details`
7. Confirm the Nickname, Age Range, and Country fields are rendered with options populated
8. Select "Canada" as a country and confirm the Province field appears
9. Trigger the "Continue" button and confirm a "Loading..." message shows up while all fields are disabled
10. Confirm the `Response: ...` output appears in the browser console and the page redirects to `/onboard/step-1`

## Screenshots

**`InfoSecondary` page on initial render, with fields empty and no visible Province selector:**

![image](https://github.com/TakingITGlobal/commit2act/assets/1672105/18870cb7-74c1-455d-bfd4-6d85a612e384)

**Fields filled-out with Canada as the country and the Province selector now visible:**

![image](https://github.com/TakingITGlobal/commit2act/assets/1672105/8e3345b3-5a70-4ec8-abd9-8c8040a9d4ed)

**Form submitted with "Loading..." text visible:**

![image](https://github.com/TakingITGlobal/commit2act/assets/1672105/5ff97692-60f0-462e-97a5-e86de75d971e)
